packages/python-packages/protocol-stub-generator/tests/test_json.py

```diff
@@ -0,0 +1,60 @@
+import json
+import os
+import pytest
+
+JSON_PATH = os.getenv("TESTJSONPATH")
+
+
+class TestJsonFormat:
+    def format_tokens(self, data):
+        d = []
+        for token in data:
+            if (
+                token["Kind"] == 4
+                or token["Kind"] == 3
+                or token["Kind"] == 0
+                or token["Kind"] == 6
+                or token["Kind"] == 7
+                or token["Kind"] == 8
+                or token["Kind"] == 10
+            ):
+                d.append(token["Value"])
+            if token["Kind"] == 2:
+                d.append(" ")
+            if token["Kind"] == 1:
+                d.append("\n")
+            else:
+                pass
+        return d
+    def test_print_out(self):
+        with open(JSON_PATH) as f:
+            data = json.load(f)
+        d = []
+        d = self.format_tokens(data["Tokens"])
+        print(*d)
+
+
+
+    def test_operation_format(self):
+        with open(JSON_PATH) as f:
+            data = json.load(f)
+        OUTPUT = data["Tokens"]
+        for t in OUTPUT:
+            if t["Value"] == "OperationGroup":
+                assert t["Kind"] == 0
+
+    def test_request_format(self):
+        r = []
+        d = []
```

Comment:
don't need to define `d` here as you assign it the return value of `format_tokens` below